nuxt-i18n.config.ts

```diff
@@ -1,5 +1,4 @@
-import { NuxtConfig } from '@nuxt/types'
-import { NuxtVueI18n } from 'nuxt-i18n/types'
+import { NuxtAppOptions, NuxtConfig } from '@nuxt/types'
```

Comment:
```suggestion
import type { NuxtConfig } from '@nuxt/types'
import type { DateTimeFormat } from 'vue-i18n'
```
`@nuxtjs/i18n` がオプションのinterfaceをNuxtConfigにマージしてくれているようなので、そこ経由で使います
`DateTimeFormat` は `dateTimeFormatsCommon` の各オプションをstringではなく定義されたリテラルとして認識させるために使います